app/models/user.rb

```diff
@@ -76,6 +77,7 @@ class User < ActiveRecord::Base
   validates_presence_of :username
   validate :username_validator, if: :username_changed?
   validates :email, presence: true, uniqueness: true
+  validates :email, format: { with: EmailValidator.email_regex }, if: :email_changed?
   validates :email, email: true, if: :should_validate_email?
```

Comment:
Is the validator (`email: true`) still needed?
And the new validator ignores the `skip_email_validation` flag. Is this intentional or should it be `if: :should_validate_email?` instead of `if: :email_changed?`?